<a href="https://colab.research.google.com/github/mgmeti/Data-Analysis/blob/main/basics.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Q1**


In [ ]:
import pandas as pd
import numpy as np

DataFrame for given dataset

In [ ]:
data = {'Name':['P. Lee','R. Jones','J. Smith','A. Patel','M. Owen','S. Green','N. Cook','W. Hands','P. Rice','F. Marsh'],
        'Age':[35, 52, 45, 70, 24, 43, 68, 77, 45, 28 ],
        'Gender': ['Female',  'Male', 'Male', 'Female', 'Male', 'Male', 'Male','Female', 'Female', 'Male'],
        'Blood Group':['ARh+', 'ORh-', 'ORh+', 'ORh-', 'ARh-', 'ORh-', 'ARh+', 'ORh-', 'ORh+', 'ORh+'],
        'Weight (kg)': [50, 115, 96, 41, 79, 109, 73, 104, 64, 136],
        'Height (m)':[1.52, 1.77, 1.83, 1.55, 1.82, 1.89, 1.76, 1.71, 1.74, 1.78],
        'Systolic Blood Pressure (mmHg)':[68, 110, 88, 76, 65, 114, 108, 107, 101, 121],
        'Diastolic Bllod Pressure (mmHg)':[112, 154, 136, 125, 105, 159, 136, 145, 132, 165],
        'Diabetes': [0, 1, 0, 0, 0, 1, 0, 1, 0, 1]}

In [ ]:
# Create DataFrame
df = pd.DataFrame(data)
df.head()

Name  Age  Gender Blood Group  Weight (kg)  Height (m)  \
0    P. Lee   35  Female        ARh+           50        1.52   
1  R. Jones   52    Male        ORh-          115        1.77   
2  J. Smith   45    Male        ORh+           96        1.83   
3  A. Patel   70  Female        ORh-           41        1.55   
4   M. Owen   24    Male        ARh-           79        1.82   

   Systolic Blood Pressure (mmHg)  Diastolic Bllod Pressure (mmHg)  Diabetes  
0                              68                              112         0  
1                             110                              154         1  
2                              88                              136         0  
3                              76                              125         0  
4                              65                              105         0

a) min-max normalisation

In [ ]:
# Function to normalize weight
def normalize_weight(weight):
    min_weight = df['Weight (kg)'].min()
    max_weight = df['Weight (kg)'].max()
    normalized_weight = (weight - min_weight) / (max_weight - min_weight)
    return round(normalized_weight, 3)

df['Normalized Weight'] = df['Weight (kg)'].apply(normalize_weight)
df.head(len(df))

Name  Age  Gender Blood Group  Weight (kg)  Height (m)  \
0    P. Lee   35  Female        ARh+           50        1.52   
1  R. Jones   52    Male        ORh-          115        1.77   
2  J. Smith   45    Male        ORh+           96        1.83   
3  A. Patel   70  Female        ORh-           41        1.55   
4   M. Owen   24    Male        ARh-           79        1.82   
5  S. Green   43    Male        ORh-          109        1.89   
6   N. Cook   68    Male        ARh+           73        1.76   
7  W. Hands   77  Female        ORh-          104        1.71   
8   P. Rice   45  Female        ORh+           64        1.74   
9  F. Marsh   28    Male        ORh+          136        1.78   

   Systolic Blood Pressure (mmHg)  Diastolic Bllod Pressure (mmHg)  Diabetes  \
0                              68                              112         0   
1                             110                              154         1   
2                              88                              136         0   
3                              76                              125         0   
4                              65                              105         0   
5                             114                              159         1   
6                             108                              136         0   
7                             107                              145         1   
8                             101                              132         0   
9                             121                              165         1   

   Normalized Weight  
0              0.095  
1              0.779  
2              0.579  
3              0.000  
4              0.400  
5              0.716  
6              0.337  
7              0.663  
8              0.242  
9              1.000

b) Binned weight
<br>
The Weight (kg) variable into three categories:
low (less than 60 kg), medium (60-100 kg), and high (greater than 100 kg).

In [ ]:
# Function to bin weight
def bin_weight(weight):
    if weight < 60:
        return 'Low'
    elif weight <= 100:
        return 'Medium'
    else:
        return 'High'

df['Binned Weight'] = df['Weight (kg)'].apply(bin_weight)
df.head(len(df))

Name  Age  Gender Blood Group  Weight (kg)  Height (m)  \
0    P. Lee   35  Female        ARh+           50        1.52   
1  R. Jones   52    Male        ORh-          115        1.77   
2  J. Smith   45    Male        ORh+           96        1.83   
3  A. Patel   70  Female        ORh-           41        1.55   
4   M. Owen   24    Male        ARh-           79        1.82   
5  S. Green   43    Male        ORh-          109        1.89   
6   N. Cook   68    Male        ARh+           73        1.76   
7  W. Hands   77  Female        ORh-          104        1.71   
8   P. Rice   45  Female        ORh+           64        1.74   
9  F. Marsh   28    Male        ORh+          136        1.78   

   Systolic Blood Pressure (mmHg)  Diastolic Bllod Pressure (mmHg)  Diabetes  \
0                              68                              112         0   
1                             110                              154         1   
2                              88                              136         0   
3                              76                              125         0   
4                              65                              105         0   
5                             114                              159         1   
6                             108                              136         0   
7                             107                              145         1   
8                             101                              132         0   
9                             121                              165         1   

   Normalized Weight Binned Weight  
0              0.095           Low  
1              0.779          High  
2              0.579        Medium  
3              0.000           Low  
4              0.400        Medium  
5              0.716          High  
6              0.337        Medium  
7              0.663          High  
8              0.242        Medium  
9              1.000          High

c) Body mass index using formula

In [ ]:
# Function to calculate BMI
def calculate_bmi(row):
    weight = row['Weight (kg)']
    height = row['Height (m)']
    bmi = weight / (height ** 2)
    return bmi

df['BMI'] = df.apply(calculate_bmi, axis=1)
df.head(len(df))


Name  Age  Gender Blood Group  Weight (kg)  Height (m)  \
0    P. Lee   35  Female        ARh+           50        1.52   
1  R. Jones   52    Male        ORh-          115        1.77   
2  J. Smith   45    Male        ORh+           96        1.83   
3  A. Patel   70  Female        ORh-           41        1.55   
4   M. Owen   24    Male        ARh-           79        1.82   
5  S. Green   43    Male        ORh-          109        1.89   
6   N. Cook   68    Male        ARh+           73        1.76   
7  W. Hands   77  Female        ORh-          104        1.71   
8   P. Rice   45  Female        ORh+           64        1.74   
9  F. Marsh   28    Male        ORh+          136        1.78   

   Systolic Blood Pressure (mmHg)  Diastolic Bllod Pressure (mmHg)  Diabetes  \
0                              68                              112         0   
1                             110                              154         1   
2                              88                              136         0   
3                              76                              125         0   
4                              65                              105         0   
5                             114                              159         1   
6                             108                              136         0   
7                             107                              145         1   
8                             101                              132         0   
9                             121                              165         1   

   Normalized Weight Binned Weight        BMI  
0              0.095           Low  21.641274  
1              0.779          High  36.707204  
2              0.579        Medium  28.666129  
3              0.000           Low  17.065557  
4              0.400        Medium  23.849777  
5              0.716          High  30.514263  
6              0.337        Medium  23.566632  
7              0.663          High  35.566499  
8              0.242        Medium  21.138856  
9              1.000          High  42.923873

### Q2

Create
a DataFrame from the data.

In [ ]:

data = {
    'x': [92, 145, 30, 70, 75, 105, 110, 108, 45, 50, 160, 155, 180, 190, 63, 85, 130, 132],
    'y': [6.3, 7.8, 3.0, 5.5, 6.5, 5.5, 6.5, 8.0, 4.0, 5.0, 7.5, 9.0, 8.6, 10.0, 4.2, 4.9, 6, 7]
}

df = pd.DataFrame(data)
df.head(len(df))

x     y
0    92   6.3
1   145   7.8
2    30   3.0
3    70   5.5
4    75   6.5
5   105   5.5
6   110   6.5
7   108   8.0
8    45   4.0
9    50   5.0
10  160   7.5
11  155   9.0
12  180   8.6
13  190  10.0
14   63   4.2
15   85   4.9
16  130   6.0
17  132   7.0

In [ ]:
# Calculate the means
mean_x = df['x'].mean()
mean_y = df['y'].mean()

# Calculate the deviations from the means
df['x - x_bar'] = df['x'] - mean_x
df['y - y_bar'] = df['y'] - mean_y

# Calculate the product of deviations
df['(x - x_bar)(y - y_bar)'] = df['x - x_bar'] * df['y - y_bar']
df.head(len(df))

x     y  x - x_bar  y - y_bar  (x - x_bar)(y - y_bar)
0    92   6.3 -14.944444  -0.105556                1.577469
1   145   7.8  38.055556   1.394444               53.066358
2    30   3.0 -76.944444  -3.405556              262.038580
3    70   5.5 -36.944444  -0.905556               33.455247
4    75   6.5 -31.944444   0.094444               -3.016975
5   105   5.5  -1.944444  -0.905556                1.760802
6   110   6.5   3.055556   0.094444                0.288580
7   108   8.0   1.055556   1.594444                1.683025
8    45   4.0 -61.944444  -2.405556              149.010802
9    50   5.0 -56.944444  -1.405556               80.038580
10  160   7.5  53.055556   1.094444               58.066358
11  155   9.0  48.055556   2.594444              124.677469
12  180   8.6  73.055556   2.194444              160.316358
13  190  10.0  83.055556   3.594444              298.538580
14   63   4.2 -43.944444  -2.205556               96.921914
15   85   4.9 -21.944444  -1.505556               33.038580
16  130   6.0  23.055556  -0.405556               -9.350309
17  132   7.0  25.055556   0.594444               14.894136

The correlation coefficient

In [ ]:
# Compute the correlation coefficient (r)
numerator = df['(x - x_bar)(y - y_bar)'].sum()
denominator = (len(df) - 1) * np.std(df['x']) * np.std(df['y'])
r = numerator / denominator

# Display the correlation coefficient (r)
print(f"Correlation coefficient (r): {r}")

Correlation coefficient (r): 0.9609465068665225


In [ ]:
# Determine if x and y are correlated
if abs(r) > 0.5:
    print("The attributes x and y are correlated.")
else:
    print("The attributes x and y are not correlated.")

The attributes x and y are correlated.


The correlation coefficient (r) between variables x and y was calculated to be 0.96. <br>This indicates a strong positive correlation between the two variables. The high value of r suggests that as the values of x increase, the values of y also tend to increase consistently. <br>Thus, there is a significant and positive linear relationship between x and y in the given dataset.